test_scripts/Smoke/API/022_SubscribeButton_PositiveCase.lua

```diff
@@ -0,0 +1,92 @@
+---------------------------------------------------------------------------------------------------
+-- User story: Smoke
+-- Use case: SubscribeButton
+-- Item: Happy path
+--
+-- Requirement summary:
+-- [SubscribeButton] SUCCESS: getting SUCCESS:SubscribeButton()
+--
+-- Description:
+-- Mobile application sends valid SubscribeButton request and gets SubscribeButton "SUCCESS" response from SDL
+
+-- Pre-conditions:
+-- a. HMI and SDL are started
+-- b. appID is registered and activated on SDL
+-- c. appID is currently in Background, Full or Limited HMI level
+
+-- Steps:
+-- appID requests SubscribeButton with valid parameters
+
+-- Expected:
+-- SDL validates parameters of the request
+-- SDL checks if Buttons interface is available on HMI
+-- SDL checks if SubscribeButton is allowed by Policies
+-- SDL checks if all parameters are allowed by Policies
+-- SDL sends the Buttons notificaton to HMI
+-- SDL responds with (resultCode: SUCCESS, success:true) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonSmoke = require('test_scripts/Smoke/commonSmoke')
+
+--[[ Local Variables ]]
+local isMediaApplication = commonSmoke.GetAppMediaStatus()
+local buttonName = {
+  "OK",
+  "SEEKLEFT",
+  "SEEKRIGHT",
+  "TUNEUP",
+  "TUNEDOWN",
+  "PRESET_0",
+  "PRESET_1",
+  "PRESET_2",
+  "PRESET_3",
+  "PRESET_4",
+  "PRESET_5",
+  "PRESET_6",
+  "PRESET_7",
+  "PRESET_8"
+}
+
+--[[ Local Functions ]]
+local function subscribeButton(pButName, self)
+  local cid = self.mobileSession1:SendRPC("SubscribeButton", { buttonName = pButName })
+  local function checkResults(pCid, pSuccess, pResultCode)
+    self.mobileSession1:ExpectResponse(pCid, { success = pSuccess, resultCode = pResultCode })
+    if pResultCode == "SUCCESS" then
+      self.mobileSession1:ExpectNotification("OnHashChange")
+      :Times(1)
+    else
+      self.mobileSession1:ExpectNotification("OnHashChange")
+      :Times(0)
+    end
+  end
+  if false == isMediaApplication and
+  ("SEEKLEFT" == pButName or
+    "SEEKRIGHT" == pButName or
+    "TUNEUP" == pButName or
+    "TUNEDOWN" == pButName) then
```

Comment:
@GetmanetsIrina propose to create in commonSmoke some function like `isButtonNameForMedia` with boolean result and move this long condition there. Then you could just use this function in all if's in both scripts like:
```
if (false == isMediaApplication) and (false == commonSmoke.isButtonNameForMedia(pButName))
OR
if (true == isMediaApplication) or (false == isMediaApplication and false == commonSmoke.isButtonNameForMedia(pButName))
```

Reply:
@GetmanetsIrina Existance of such logic in script is bad practice. I would propose to create separated tests with simplified logic and applicable parameters for mobile apps with `isMediaApplication = true` and `isMediaApplication = false` instead of guessing about application type in script.

Reply:
@AKalinich-Luxoft , according to internal discussion with @aderiabin  and @dboltovskyi for non-media application will be created separated scripts and test set, so code is removed.